tests/components/modbus/test_modbus.py

```diff
@@ -50,23 +103,397 @@
             CONF_PARITY: "E",
             CONF_STOPBITS: 1,
         },
+        {
+            CONF_TYPE: "serial",
+            CONF_BAUDRATE: 9600,
+            CONF_BYTESIZE: 8,
+            CONF_METHOD: "rtu",
+            CONF_PORT: "usb01",
+            CONF_PARITY: "E",
+            CONF_STOPBITS: 1,
+            CONF_NAME: "modbusTest",
+            CONF_TIMEOUT: 30,
+            CONF_DELAY: 10,
+        },
     ],
 )
-async def test_config_modbus(hass, do_discovery, do_options, do_config):
+async def test_config_modbus(hass, do_config):
     """Run test for modbus."""
+
+    await _config_helper(hass, do_config)
+    assert DOMAIN in hass.data
+    assert len(hass.data[DOMAIN]) == 1
+
+
+async def test_config_multiple_modbus(hass):
+    """Run test for multiple modbus."""
+
+    do_config = [
+        {
+            CONF_TYPE: "tcp",
+            CONF_HOST: "modbusTestHost",
+            CONF_PORT: 5501,
+            CONF_NAME: "modbusTest1",
+        },
+        {
+            CONF_TYPE: "tcp",
+            CONF_HOST: "modbusTestHost",
+            CONF_PORT: 5501,
+            CONF_NAME: "modbusTest2",
+        },
+        {
+            CONF_TYPE: "serial",
+            CONF_BAUDRATE: 9600,
+            CONF_BYTESIZE: 8,
+            CONF_METHOD: "rtu",
+            CONF_PORT: "usb01",
+            CONF_PARITY: "E",
+            CONF_STOPBITS: 1,
+            CONF_NAME: "modbusTest3",
+        },
+    ]
+
+    await _config_helper(hass, do_config)
+    assert DOMAIN in hass.data
+    assert len(hass.data[DOMAIN]) == len(do_config)
+
+
+@pytest.fixture()
+def modbus_hub():
+    """Return class obj configured."""
+
+    config_hub = {
+        CONF_NAME: DEFAULT_HUB,
+        CONF_TYPE: "tcp",
+        CONF_HOST: "modbusTest",
+        CONF_PORT: 5001,
+        CONF_DELAY: 1,
+        CONF_TIMEOUT: 1,
+    }
+    hub = ModbusHub(config_hub)
+    assert hub.name == config_hub[CONF_NAME]
+    return hub
+
+
+async def test_class_ModbusHub_logger(hass, caplog, modbus_hub):
+    """Run general test of class modbusHub."""
+
+    caplog.set_level(logging.DEBUG)
+    e = ModbusException("test logger")
+    caplog.clear()
+    modbus_hub._log_error(e)
```

Comment:
Is this really something we need to test? We should avoid accessing things that may change and that we don't really care about. Private methods fall into that category.

Reply:
You are quite right, I simply got carried away when writing tests :-) Removed.